lib/dynamo.js

```diff
@@ -1,182 +1,174 @@
-    ReturnValues: 'ALL_OLD',
+    ReturnValues: "ALL_OLD",
     TableName: process.env.DDB_TABLE,
   };
 
   // compress payload data
   if (payload) {
-    const compressed = await exports.deflate(payload);
-    params.AttributeUpdates.payload = { Action: 'PUT', Value: { B: compressed } };
+    params.AttributeUpdates.payload = {
+      Action: "PUT",
+      Value: { B: await deflate(payload) },
+    };
   }
 
-  // stringify json data
-  if (segments && segments.length) {
-    const strings = (segments || []).map(s => s.toString());
-    params.AttributeUpdates.segments = { Action: 'ADD', Value: { SS: strings } };
+  // stringify segments array
+  if (segments?.length) {
+    params.AttributeUpdates.segments = {
+      Action: "ADD",
+      Value: { SS: (segments || []).map((s) => s.toString()) },
+    };
   }
 
   // optional extra positioning data about the segments
   if (extras) {
-    params.AttributeUpdates.extras = { Action: 'PUT', Value: { S: JSON.stringify(extras) } };
+    params.AttributeUpdates.extras = {
+      Action: "PUT",
+      Value: { S: JSON.stringify(extras) },
+    };
   }
 
   // optional expiration
   const ttl = +process.env.DDB_TTL;
   if (ttl > 0) {
-    const expires = Math.round(new Date().getTime() / 1000) + ttl;
-    params.AttributeUpdates.expiration = { Action: 'PUT', Value: { N: expires.toString() } };
+    const expires = Math.round(Date.now() / 1000) + ttl;
+    params.AttributeUpdates.expiration = {
+      Action: "PUT",
+      Value: { N: expires.toString() },
+    };
   }
 
-  return params;
-};
+  const command = new UpdateItemCommand(params);
+  const result = await (client || (await getClient())).send(command);
+  return new DynamoData({ id, payload, segments, extras, result });
+}
 
 /**
- * Return the id, payload, and any "new" segments
+ * Add timestamps to the frequency capping table, for a campaign+listener
  */
-exports.updateResult = async (id, setPayload, setSegments, setExtras, result) => {
-  const oldPayload =
-    result.Attributes && result.Attributes.payload ? result.Attributes.payload.B : null;
-  const oldSegments =
-    result.Attributes && result.Attributes.segments ? result.Attributes.segments.SS : [];
-  const oldExtras =
-    result.Attributes && result.Attributes.extras ? result.Attributes.extras.S : null;
-
-  // only inflate the payload if necessary
-  const newPayload = setPayload || (oldPayload && (await exports.inflate(oldPayload))) || null;
-
-  // merge extras into the payload
-  if (newPayload && setExtras) {
-    Object.keys(setExtras).forEach(key => (newPayload[key] = setExtras[key]));
-  } else if (newPayload && oldExtras) {
-    try {
-      const extras = JSON.parse(oldExtras);
-      Object.keys(extras).forEach(key => (newPayload[key] = extras[key]));
-    } catch (err) {
-      logger.warn(`Error json parsing extras: ${err}`, { extras: oldExtras });
-    }
-  }
+export async function addFrequency({ client, listener, campaign, timestamp, maxSeconds } = {}) {
```

Comment:
Above this line, this file is functionally the same.  Just syntax changes and moving a few things into `lib/dynamo-data.js` helper class.

From here down, I ported the actual DDB commands out of `lib/inputs/dovetail-frequency.js`.  Just so all the DDB commands are in one place.